cli: productionize debug check-store

tbg opened the pull request:
In light of the stats inconsistency seen in [clearrange], we want to be
stricter about verifying the stats in nightly testing. This commit makes
sure `./cockroach debug check-store` is fast enough to do so:

On a ~71GB fully compacted store directory it reliably takes well below
two minutes (on GCE local SSD).

[clearrange]: #38720 (comment)

Release note (performance improvement): The `./cockroach debug check-store` command is now faster.

tbg committed:
cli: move check-store to own file

Release note: None

tbg commented:
@andreimatei I think Ben is out, could you help me get this in?

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-:-Lrt0v3K2GjYCdPFi1nT:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei, @bdarnell, and @tbg)

___
*[pkg/cli/debug_check_store.go, line 53 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrszrek3-_efd6BOKOg:-Lrszrek3-_efd6BOKOh:bsdic60) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L53)):*
> ```Go
> var errCheckFoundProblem = errors.New("check-store found problems")
> 
> func runDebugCheckStoreCmd(cmd *cobra.Command, args []string) error {
> ```

take in a ctx pls
___
*[pkg/cli/debug_check_store.go, line 58 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-LrszzOpDby3QG7vMx99:-LrszzOpDby3QG7vMx9A:b-ucf1of) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L58)):*
> ```Go
> 	// At time of writing, this takes around ~110s for 71GB (1k warehouse TPCC
> 	// fully compacted) on local SSD. This is quite fast, well north of 600MB/s.
> 	err := checkStoreRangeStats(ctx, args[0], fmt.Println)
> ```

nit: pls to `dir := args[0]`
___
*[pkg/cli/debug_check_store.go, line 138 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt-Bt9AWa0ghFd5Az5:-Lrt-Bt9AWa0ghFd5Az6:bjhf38g) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L138)):*
> ```Go
> }
> 
> func checkStoreRangeStats(
> ```

pls comment the function

s/dir/storeDir

pls comment println. What is it called with? Better yet, let's make a typed interface for returning these results, or a typed channel if they need to be returned async.
___
*[pkg/cli/debug_check_store.go, line 144 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt-oKi4T4OLSIvCmVT:-Lrt-oKjAVBgk5w70RBI:b-2tgdx8) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L144)):*
> ```Go
> 	defer stopper.Stop(ctx)
> 
> 	eng, err := OpenExistingStore(dir, stopper, true /* readOnly */)
> ```

nit: let's get rid of the stopper argument to `OpenExistingStore()`. Make it return the cleanup fn.
___
*[pkg/cli/debug_check_store.go, line 178 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt0cWk7N7zCOpHOXj8:-Lrt0cWk7N7zCOpHOXj9:b-8ddeoy) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L178)):*
> ```Go
> 	}()
> 
> 	var out []string
> ```

do we do anything with this `out` other than check if it's empty?


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-:-Ls1PRKj6jCz_esgpccS:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @bdarnell)

___
*[pkg/cli/debug_check_store.go, line 53 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrszrek3-_efd6BOKOg:-Ls1MN5h3xbK8imq3lAG:bmxqvoq) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L53)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

take in a ctx pls
</blockquote></details>

This is the cobra-mandated signature.
___
*[pkg/cli/debug_check_store.go, line 138 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt-Bt9AWa0ghFd5Az5:-Ls1MYPP652xk2tvg75U:b-jub9pn) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L138)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

pls comment the function

s/dir/storeDir

pls comment println. What is it called with? Better yet, let's make a typed interface for returning these results, or a typed channel if they need to be returned async.
</blockquote></details>

I added a comment that this is just `fmt.Println` outside of tests. I don't want to hold back results until the end of the run since this is a long-running op, and I don't feel that adding more complexity (channel etc) is worth it yet.
___
*[pkg/cli/debug_check_store.go, line 144 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt-oKi4T4OLSIvCmVT:-Ls1PMG2AShDBbFVw3B3:bxw1bea) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L144)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: let's get rid of the stopper argument to `OpenExistingStore()`. Make it return the cleanup fn.
</blockquote></details>

I looked but it's adding more to this PR than I like.
___
*[pkg/cli/debug_check_store.go, line 178 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/41805#-Lrt0cWk7N7zCOpHOXj8:-Ls1Nwtx8V4O1OMLiDjh:b-4c5ezp) ([raw file](https://github.com/cockroachdb/cockroach/blob/ca2a89ae7268f7e4c7c916973a864aba88613773/pkg/cli/debug_check_store.go#L178)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

do we do anything with this `out` other than check if it's empty?
</blockquote></details>

Good catch, done.


<!-- Sent from Reviewable.io -->

tbg committed:
cli: productionize debug check-store

In light of the stats inconsistency seen in [clearrange], we want to be
stricter about verifying the stats in nightly testing. This commit makes
sure `./cockroach debug check-store` is fast enough to do so:

On a ~71GB fully compacted store directory it reliably takes well below
two minutes (on GCE local SSD).

[clearrange]: https://github.com/cockroachdb/cockroach/issues/38720#issuecomment-538997993

Release note (performance improvement): The `./cockroach debug check-store` command is now faster.
